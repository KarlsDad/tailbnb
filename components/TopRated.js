import TopRatedCard from '../components/TopRatedCard';

import img1 from '../public/adventures1.jpg';
import ShowAll from './ShowAll';

export default () => {
  return (
    <div className='xl:max-w-5.5xl lg:max-w-5.5xl mx-auto mt-14 mb-20'>
      <h3 className='text-2xl font-semibold py-3 mt-5'>
        Top-rated experiences
      </h3>
      <p className='mb-6'>
        Book activities led by local hosts on your next trip.
      </p>
      <div className='flex items-start justify-start'>
        <div className='xl:w-1/4 lg:w-1/2'>
          <TopRatedCard
            img={img1}
            location='Italy'
            title='Luca & Lorenzo, cooking in Chianti hills'
            price='172'
          />
        </div>
        <div className='xl:w-1/4 lg:w-1/2'>
          <TopRatedCard
            img={img1}
            location='Italy'
            title='Luca & Lorenzo, cooking in Chianti hills'
            price='172'
          />
        </div>
        <div className='xl:w-1/4 lg:w-1/2'>
          <TopRatedCard
            img={img1}
            location='Italy'
            title='Luca & Lorenzo, cooking in Chianti hills'
            price='172'
          />
        </div>
        <div className='xl:w-1/4 lg:w-1/2'>
          <TopRatedCard
            img={img1}
            location='Italy'
            title='Luca & Lorenzo, cooking in Chianti hills'
            price='172'
          />
        </div>
      </div>
      <ShowAll title='Show all experiences' />
    </div>
  );
};
