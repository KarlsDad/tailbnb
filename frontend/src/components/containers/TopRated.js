import React, { useState } from 'react';
import { useQuery } from '@apollo/react-hooks';
import { gql } from 'apollo-boost';
import PulseLoader from 'react-spinners/PulseLoader';
import Media from 'react-media';

// Components
import TopRatedCard from '../presentational/TopRatedCard';
import ShowAll from '../ShowAll';

const GET_EXPERIENCES = gql`
  query {
    experiences {
      id
      title
      cost
      ratings
      reviews
      ratings
      location
      imglow
      img
    }
  }
`;

export default () => {
  const { loading, error, data } = useQuery(GET_EXPERIENCES);
  const [card, setCard] = useState(0);

  const renderContent = (data, number) => {
    var content = [];
    for (let i = 0; i < number; i++) {
      content.push(
        <div className='w-1/2 md:w-1/3 lg:w-1/4 xl:w-1/5 2xl:w-1/6 pb-5'>
          <TopRatedCard
            key={data[i].id}
            id={data[i].id}
            img={data[i].img}
            imglow={data[i].imglow}
            country={data[i].country}
            title={data[i].title}
            cost={data[i].cost}
            period={data[i].period}
            ratings={data[i].ratings}
            reviews={data[i].reviews}
          />
        </div>
      );
    }
    return content;
  };

  if (loading) {
    return (
      <div className='flex justify-center items-center w-full py-20'>
        <PulseLoader size={10} color={'#008489'} />
      </div>
    );
  }

  if (error) return `Error! ${error.message}`;

  return (
    <>
      <div className='flex items-start justify-start flex-wrap w-full'>
        {data ? (
          <>
            <Media queries={{ xs: '(min-width: 0px) and (max-width: 739px)' }}>
              {(matches) =>
                matches.xs
                  ? (setCard(4), renderContent(data.experiences, card))
                  : null
              }
            </Media>
            <Media
              queries={{ small: '(min-width: 740px) and (max-width: 987px)' }}>
              {(matches) =>
                matches.small
                  ? (setCard(3), renderContent(data.experiences, card))
                  : null
              }
            </Media>
            <Media
              queries={{
                large: '(min-width: 988px) and (max-width: 1299px)'
              }}>
              {(matches) =>
                matches.large
                  ? (setCard(4), renderContent(data.experiences, card))
                  : null
              }
            </Media>
            <Media
              queries={{
                xl: '(min-width: 1300px) and (max-width: 1529px)'
              }}>
              {(matches) =>
                matches.xl
                  ? (setCard(5), renderContent(data.experiences, card))
                  : null
              }
            </Media>
            <Media
              queries={{
                xl: '(min-width: 1530px)'
              }}>
              {(matches) =>
                matches.xl
                  ? (setCard(6), renderContent(data.experiences, card))
                  : null
              }
            </Media>
          </>
        ) : null}
      </div>
      <ShowAll title='Show all experiences' />
    </>
  );
};
