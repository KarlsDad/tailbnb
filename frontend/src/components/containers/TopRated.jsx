import * as React from 'react';
import { useQuery } from '@apollo/react-hooks';
import { gql } from 'apollo-boost';
import PulseLoader from 'react-spinners/PulseLoader';
import withSizes from 'react-sizes';

// Utils
import { RenderSkeletonVertical } from '../../util/RenderSkeleton';

// Components
import { TopRatedCard } from '../functions/TopRatedCard';
import { ShowAll } from '../ShowAll';

// Wrapper
import { Section } from '../wrapper/Section';

const GET_EXPERIENCES = gql`
  query {
    experiences {
      id
      title
      cost
      ratings
      reviews
      location
      img
    }
  }
`;

const mapSizesToProps = ({ width }) => ({
  isMobile: width < 767,
  isTablet: width > 767 && width < 1023,
  isLaptop: width > 1023 && width < 1279,
  isDesktop: width > 1279 && width < 1529,
  isLargeDesktop: width > 1529
});

const renderContent = (data, number) => {
  var content = [];

  for (let i = 0; i < number; i++) {
    content.push(
      <div className='pb-5'>
        <TopRatedCard
          key={i}
          id={data?.experiences[i].id}
          img={data?.experiences[i].img}
          title={data?.experiences[i].title}
          cost={data?.experiences[i].cost}
          ratings={data?.experiences[i].ratings}
          reviews={data?.experiences[i].reviews}
          location={data?.experiences[i].location}
        />
      </div>
    );
  }
  return content;
};

// interface Experience {
//   id: string;
//   title: string;
//   cost: number;
//   ratings: number;
//   reviews: number;
//   location: string;
//   img: string;
// }

// interface ExperienceData {
//   experiences: Experience[];
// }

export const TopRated = ({
  isMobile,
  isTablet,
  isLaptop,
  isDesktop,
  isLargeDesktop
}) => {
  const { loading, error, data } = useQuery(GET_EXPERIENCES);

  if (error) return `Error! ${error.message}`;

  return (
    <>
      <Section
        title='Top-rated experiences'
        phrase='Book activities led by local hosts on your next trip.'>
        {loading ? (
          <div className='grid gap-3 2xl:grid-cols-6 xl:grid-cols-5 lg:grid-cols-4 md:grid-cols-3 grid-cols-2 w-full h-88'>
            {isMobile ? RenderSkeletonVertical(4) : null}
            {isTablet ? RenderSkeletonVertical(3) : null}
            {isLaptop ? RenderSkeletonVertical(4) : null}
            {isDesktop ? RenderSkeletonVertical(5) : null}
            {isLargeDesktop ? RenderSkeletonVertical(6) : null}
          </div>
        ) : (
          data && (
            <div className='grid gap-3 2xl:grid-cols-6 xl:grid-cols-5 lg:grid-cols-4 md:grid-cols-3 grid-cols-2 w-full'>
              {isMobile ? renderContent(data, 4) : null}
              {isTablet ? renderContent(data, 3) : null}
              {isLaptop ? renderContent(data, 4) : null}
              {isDesktop ? renderContent(data, 5) : null}
              {isLargeDesktop ? renderContent(data, 6) : null}
            </div>
          )
        )}

        <ShowAll title='Show all experiences' />
      </Section>
    </>
  );
};

export default withSizes(mapSizesToProps)(TopRated);
