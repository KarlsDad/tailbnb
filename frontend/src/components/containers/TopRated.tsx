import * as React from 'react';
import { useState } from 'react';
import { useQuery } from '@apollo/react-hooks';
import { gql } from 'apollo-boost';
import PulseLoader from 'react-spinners/PulseLoader';
import Media from 'react-media';

// Components
import { TopRatedCard } from '../functions/TopRatedCard';
import { ShowAll } from '../ShowAll';

// Wrapper
import { Section } from '../wrapper/Section';

const GET_EXPERIENCES = gql`
  query {
    experiences {
      id
      title
      cost
      ratings
      reviews
      location
      img
    }
  }
`;

interface Experience {
  id: string;
  title: string;
  cost: number;
  ratings: number;
  reviews: number;
  location: string;
  img: string;
}

interface ExperienceData {
  experiences: Experience[];
}

export const TopRated: any = () => {
  const { loading, error, data } = useQuery<ExperienceData>(GET_EXPERIENCES);
  const [card, setCard] = useState<number>(0);

  const renderContent = (data: ExperienceData, number: number) => {
    var content: any[] = [];

    for (let i = 0; i < number; i++) {
      content.push(
        <div className='w-1/2 md:w-1/3 lg:w-1/4 2xl:w-1/6 pb-5'>
          <TopRatedCard
            key={i}
            id={data?.experiences[i].id}
            img={data?.experiences[i].img}
            title={data?.experiences[i].title}
            cost={data?.experiences[i].cost}
            ratings={data?.experiences[i].ratings}
            reviews={data?.experiences[i].reviews}
            location={data?.experiences[i].location}
          />
        </div>
      );
    }
    return content;
  };

  if (error) return `Error! ${error.message}`;

  return (
    <>
      <Section
        title='Top-rated experiences'
        phrase='Book activities led by local hosts on your next trip.'>
        <div className='flex items-start justify-start flex-wrap w-full'>
          {loading ? (
            <div className='flex justify-center items-center w-full py-20'>
              <PulseLoader size={10} color={'#008489'} />
            </div>
          ) : (
            data && (
              <>
                <Media
                  queries={{ xs: '(min-width: 0px) and (max-width: 739px)' }}>
                  {(matches) =>
                    matches.xs ? (setCard(1), renderContent(data, card)) : null
                  }
                </Media>
                <Media
                  queries={{
                    small: '(min-width: 740px) and (max-width: 987px)'
                  }}>
                  {(matches) =>
                    matches.small
                      ? (setCard(1), renderContent(data, card))
                      : null
                  }
                </Media>
                <Media
                  queries={{
                    large: '(min-width: 988px) and (max-width: 1299px)'
                  }}>
                  {(matches) =>
                    matches.large
                      ? (setCard(4), renderContent(data, card))
                      : null
                  }
                </Media>
                <Media
                  queries={{
                    xl: '(min-width: 1300px) and (max-width: 1529px)'
                  }}>
                  {(matches) =>
                    matches.xl ? (setCard(4), renderContent(data, card)) : null
                  }
                </Media>
                <Media
                  queries={{
                    twoxl: '(min-width: 1530px)'
                  }}>
                  {(matches) =>
                    matches.twoxl
                      ? (setCard(6), renderContent(data, card))
                      : null
                  }
                </Media>
              </>
            )
          )}
        </div>
        <ShowAll title='Show all experiences' />
      </Section>
    </>
  );
};
