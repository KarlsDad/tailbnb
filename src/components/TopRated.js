import React from 'react';

// Components
import TopRatedCard from '../components/TopRatedCard';
import ShowAll from './ShowAll';

// Images
import experience1 from '../../public/img/high/experience-1.jpg';
import experience2 from '../../public/img/high/experience-2.jpg';
import experience3 from '../../public/img/high/experience-3.jpg';
import experience4 from '../../public/img/high/experience-4.jpg';

export default () => {
  const toprateds = [
    {
      img: experience1,
      location: 'Italy',
      title: 'Luca & Lorenzo, cooking in Chianti hills',
      price: 172,
      reviews: 1765
    },
    {
      img: experience2,
      location: 'Italy',
      title: 'Truffle experience Rome',
      price: 148,
      reviews: 621
    },
    {
      img: experience3,
      location: 'Japan',
      title: 'Feeling Samurai Soul',
      price: 119,
      reviews: 1047
    },
    {
      img: experience4,
      location: 'Italy',
      title: 'Lisbon best flavors',
      price: 90,
      reviews: 3733
    }
  ];
  return (
    <>
      <div className='flex items-start justify-start flex-wrap w-full'>
        {toprateds.map(({ img, location, title, price, reviews }) => {
          <div className='md:w-1/4 sm:w-1/3 w-1/2 pb-5'>
            <TopRatedCard
              img={img}
              location={location}
              title={title}
              price={price}
              reviews={reviews}
            />
          </div>;
        })}
      </div>
      <ShowAll title='Show all experiences' />
    </>
  );
};
