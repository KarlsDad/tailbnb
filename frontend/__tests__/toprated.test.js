import React from 'react';
import { shallow } from 'enzyme';

// Component
import TopRated from '../src/components/TopRated';

describe('TopRated', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<TopRated />);
    expect(component).toMatchSnapshot();
  });
});
