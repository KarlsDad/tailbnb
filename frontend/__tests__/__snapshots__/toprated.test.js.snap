// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopRated should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
